packages/reference/src/common/EntityStore.tsx

```diff
@@ -92,8 +92,43 @@ type EntityQueryKey = [
   environmentId: string
 ];
 
+type ResourceProviderQueryKey = [
+  ident: 'ResourceProvider',
+  organizationId: string,
+  appDefinitionId: string
+];
+
 type ScheduledActionsQueryKey = ['scheduled-actions', ...EntityQueryKey];
 
+export type FunctionInvocationErrorResponse = {
+  status: number;
+  statusText: string;
+  message:
+    | 'Response payload of the Contentful Function is invalid'
+    | 'An error occurred while executing the Contentful Function code';
+  request: {
+    url: string;
+    headers: Record<string, string>;
+    method: string;
+  };
+};
+
+function isFunctionInvocationErrorResponse(
+  response: unknown
+): response is FunctionInvocationErrorResponse {
+  const functionInvocationErrorMessages = [
+    'An error occurred while executing the Contentful Function code',
+    'Response payload of the Contentful Function is invalid',
+  ];
+  return (
+    response !== null &&
+    typeof response === 'object' &&
+    'message' in response &&
+    typeof response.message === 'string' &&
```

Comment:
Why not just:
```ts
return (typeof response?.message === 'string' && functionInvocationErrorMessages.includes(response.message));
```

Reply:
The `response` param was `unknown` in a previous version of this code, so I had to do all those checks. I'll change that to make it simpler now that we're doing an `instanceof Error` check before this function is called.